presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -100,11 +113,13 @@ public Response updateState(NodeState state)
     @Produces(APPLICATION_JSON)
     public NodeState getServerState()
     {
-        if (shutdownHandler.isShutdownRequested()) {
-            return SHUTTING_DOWN;
-        }
-        else {
-            return ACTIVE;
+        synchronized (shutdownHandler) {
+            if (shutdownHandler.isShutdownRequested()) {
```

Comment:
`isShutdownRequested` is already synchronized, why do you need the synch. block?